docs/Coding/macros.rst

```diff
@@ -0,0 +1,121 @@
+Feature Macros
```

Comment:
I don't think this file should be in the documentation (sorry). It was more of a WIP document while we were cleaning up the feature macros. The list is now much smaller:

```
#elif defined(ENABLE_NLS) && defined(HAVE_STRCASECOLL)
#elif defined(ENABLE_NLS) && defined(HAVE_STRICOLL)
#ifdef FREECIV_HAVE_BZ2
#ifdef FREECIV_HAVE_LIBINTL_H
#ifdef FREECIV_HAVE_LZMA
#ifdef FREECIV_HAVE_ZSTD
#ifdef HAVE_AT_QUICK_EXIT
#ifdef HAVE_CONFIG_H
#ifdef HAVE_DIRECT_H
#ifdef HAVE_MAPIMG_MAGICKWAND
#ifdef HAVE_STRING_H
#ifdef HAVE_STRINGS_H
#ifdef HAVE_SUPPRESS_APPEND
#if defined(ENABLE_NLS) && defined(HAVE__STRICOLL)
```

Some of them could still be dropped but it's nothing like what we had when we started...

It would make sense to keep the command that lists all feature macros (was quite horrible to write it), but even this isn't critical.

Reply:
I had a feeling that that document was something of a scratch pad. I'll revert.